spec/unit/banner_spec.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+require_relative "../spec_helper"
+
+module DeadEnd
+  RSpec.describe Banner do
+    it "Unmatched | banner" do
+      source = <<~EOM
+        Foo.call do |
+        end
+      EOM
+
+      invalid_obj = WhoDisSyntaxError.new(source)
+      banner = Banner.new(invalid_obj: invalid_obj)
+      expect(banner.call).to include("Unmatched `|` character detected")
+    end
+
+    it "Unmatched { banner" do
+      source = <<~EOM
+        class Cat
+          lol = {
+        end
+      EOM
+
+      invalid_obj = WhoDisSyntaxError.new(source)
+      banner = Banner.new(invalid_obj: invalid_obj)
+      expect(banner.call).to include("Unmatched `{` character detected")
+    end
+
+    it "Unmatched } banner" do
+      skip("Unsupported ruby version") unless Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("2.7")
```

Comment:
Error message for this one on 2.6 looks off for some reason:

```
➜  dead_end git:(mauro-oto/support-naked-braces-brackets-and-invert-labels) ✗ asdf local ruby 2.6.8
➜  dead_end git:(mauro-oto/support-naked-braces-brackets-and-invert-labels) ✗ ruby -wc soaprequest_deadend.rb
soaprequest_deadend.rb:2: syntax error, unexpected tSTRING_DEND
  lol = }

➜  dead_end git:(mauro-oto/support-naked-braces-brackets-and-invert-labels) ✗ asdf local ruby 2.7.4
➜  dead_end git:(mauro-oto/support-naked-braces-brackets-and-invert-labels) ✗ ruby -wc soaprequest_deadend.rb
soaprequest_deadend.rb:2: syntax error, unexpected '}'
  lol = }

➜  dead_end git:(mauro-oto/support-naked-braces-brackets-and-invert-labels) ✗ asdf local ruby 3.0.2
➜  dead_end git:(mauro-oto/support-naked-braces-brackets-and-invert-labels) ✗ ruby -wc soaprequest_deadend.rb
soaprequest_deadend.rb:2: syntax error, unexpected '}'
  lol = }
```

Reply:
I'm feeling more and more like we can't rely on the parser error messages. I'll put more of my thoughts at the top level.